internal/resources/policy/kind/quota/resource/resource_quota_policy_test.go

```diff
@@ -0,0 +1,396 @@
+	time.Sleep(2 * time.Minute)
+
+	// Test case for quota policy resource with small recipe.
+	resource.Test(t, resource.TestCase{
+		PreCheck:          testhelper.TestPreCheck(t),
+		ProviderFactories: testhelper.GetTestProviderFactories(testConfig.Provider),
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					if testConfig.ScopeHelperResources.Cluster.KubeConfigPath == "" {
+						t.Skip("KUBECONFIG env var is not set for cluster scoped quota policy acceptance test")
+					}
+				},
+				Config: testConfig.getTestQuotaPolicyResourceBasicConfigValue(scope.ClusterScope, policykindquota.SmallRecipe),
+				Check:  testConfig.checkQuotaPolicyResourceAttributes(scope.ClusterScope),
+			},
+			{
+				Config: testConfig.getTestQuotaPolicyResourceBasicConfigValue(scope.ClusterGroupScope, policykindquota.SmallRecipe),
+				Check:  testConfig.checkQuotaPolicyResourceAttributes(scope.ClusterGroupScope),
+			},
+			{
+				PreConfig: func() {
+					if testConfig.ScopeHelperResources.OrgID == "" {
+						t.Skip("ORG_ID env var is not set for organization scoped quota policy acceptance test")
+					}
+				},
+				Config: testConfig.getTestQuotaPolicyResourceBasicConfigValue(scope.OrganizationScope, policykindquota.SmallRecipe),
+				Check:  testConfig.checkQuotaPolicyResourceAttributes(scope.OrganizationScope),
+			},
+		},
+	},
+	)
+
+	t.Log("quota policy resource acceptance test complete for small recipe!")
+	time.Sleep(2 * time.Minute)
+
+	// Test case for quota policy resource with medium recipe.
+	resource.Test(t, resource.TestCase{
+		PreCheck:          testhelper.TestPreCheck(t),
+		ProviderFactories: testhelper.GetTestProviderFactories(testConfig.Provider),
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					if testConfig.ScopeHelperResources.Cluster.KubeConfigPath == "" {
+						t.Skip("KUBECONFIG env var is not set for cluster scoped quota policy acceptance test")
+					}
+				},
+				Config: testConfig.getTestQuotaPolicyResourceBasicConfigValue(scope.ClusterScope, policykindquota.MediumRecipe),
+				Check:  testConfig.checkQuotaPolicyResourceAttributes(scope.ClusterScope),
+			},
+			{
+				Config: testConfig.getTestQuotaPolicyResourceBasicConfigValue(scope.ClusterGroupScope, policykindquota.MediumRecipe),
+				Check:  testConfig.checkQuotaPolicyResourceAttributes(scope.ClusterGroupScope),
+			},
+			{
+				PreConfig: func() {
+					if testConfig.ScopeHelperResources.OrgID == "" {
+						t.Skip("ORG_ID env var is not set for organization scoped quota policy acceptance test")
+					}
+				},
+				Config: testConfig.getTestQuotaPolicyResourceBasicConfigValue(scope.OrganizationScope, policykindquota.MediumRecipe),
+				Check:  testConfig.checkQuotaPolicyResourceAttributes(scope.OrganizationScope),
+			},
+		},
+	},
+	)
+
+	t.Log("quota policy resource acceptance test complete for medium recipe!")
+	time.Sleep(2 * time.Minute)
```

Comment:
why are these sleep statements needed ?

Reply:
These ensure to give ample time to clean up the cluster/ cluster group created for a particular recipe type before starting the creation of another one to avoid conflict of helper resource- scope( cluster/ cluster group) and smooth running of acceptance tests.

Reply:
Again, this has been tested in the past for security and custom policy and since the issue was encountered, sleep was introduced in all the acceptance tests pertaining to policies where we have multiple recipes and multiple scopes unlike other resources.

Reply:
Why can't we use new cluster and cluster group names ? 
having 2 mins sleep for every test will make the tests very slow and this does not guarantee that the cluster is actually cleaned. 
This is a wrong way of writing the acceptance test and rely on sleep is not accepted

Reply:
Can you suggest a better way to avoid sleep at all? We are using helper functions to create scopes, we would then need more of those. Again, this would then have to be changed for all other kinds of policies.

Reply:
I remember commenting on this earlier as well. 
There are two ways to address this. 
1. use new name for cluster and cluster group. This avoids the block on the clean up
2. come up with helper function which will wait only for the time duration needed for cluster clean up. 

I would prefer having option 1 implemented for all the acceptance test of policy. 

Their PR can be get merged but this needs to be fixed in the very next PR. Having 2 mins sleep is very costly for each acceptance test

@sbasu31